Kitodo/src/main/java/org/kitodo/services/file/FileService.java

```diff
@@ -654,11 +653,10 @@ public URI getProcessBaseUriForExistingProcess(Process process) {
      * @return The URI of the requested location
      */
     public URI getProcessSubTypeURI(Integer processId, String processTitle, URI processDataDirectory,
-            ProcessSubType processSubType, String resourceName) throws DAOException {
+            ProcessSubType processSubType, String resourceName) {
 
         if (processDataDirectory == null) {
-            Process process = serviceManager.getProcessService().getById(processId);
-            processDataDirectory = serviceManager.getProcessService().getProcessDataDirectory(process);
+            processDataDirectory = URI.create(String.valueOf(processId));
```

Comment:
I make assumption that if this field is empty inside the index (index has also this field) inside the database it should be also empty therefore I will use id.

Reply:
Naah, sorry, you need to call the processservice, because a lot of magic is going on there. The field can be empty for processes which have been migrated from a former (2x) installation. The used process directory needs to be retrieved, which is happening in the process service.

Reply:
Naah, sorry, you need to call the processservice, because a lot of magic is going on there. The field can be empty for processes which have been migrated from a former (2x) installation. The used process directory needs to be retrieved, which is happening in the process service.